### PR TITLE
Mail Service

### PR DESCRIPTION
**Pull request by jacob-b-labs**

### CODE REVIEW

**Comment by WasilewskiJakub** [comment]
Potwierdzam pull request

File: Backend/Controllers/HealthController.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+public class HealthController : ControllerBase
+{
+    private readonly ILogger<HealthController> _logger;
+
+    public HealthController(ILogger<HealthController> logger)
+    {
+        _logger = logger;
+    }
+
+    [HttpGet(Name = "GetWeatherForecast")]
```
Comment (complikator) [inline Backend/Controllers/HealthController.cs:16]: I think it should be changed

File: Backend/Controllers/HealthController.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    public string Get()
+    {
+        return $"{DateTime.Now} OK";
```
Comment (complikator) [inline Backend/Controllers/HealthController.cs:19]: Preferably JSON instead of literal string

File: Backend/Program.cs [hunk 3]
```diff
@@ -1,25 +1,51 @@
...
 var builder = WebApplication.CreateBuilder(args);
 
-// Add services to the container.
+// builder.Services.AddDbContext<CoreDbContext>(
+//     opts => opts.UseNpgsql(builder.Configuration["DatabaseConnectionString"])
+// );
+
+// Singletons
+builder.Services.AddSingleton(new MailConfiguration(
+    builder.Configuration["MailHost"]!,
```
Comment (complikator) [inline Backend/Program.cs:12]: Maybe put all settings in a common struct thus change settings to sth like "MailServiceSettings: Host"

File: Backend/Services/MailService.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        var message = new MimeMessage();
+        message.From.Add(new MailboxAddress(cfg.Sender, cfg.SenderMail));
+        message.To.Add(new MailboxAddress(Recipient, RecipientMail));
+        message.Subject = Subject;
+
+        message.Body = new TextPart(UseHtml ? "html" : "plain") { Text = Body };
+
+        using (var client = new SmtpClient())
+        {
+            await client.ConnectAsync(cfg.Host, port, useSsl, ct);
```
Comment (complikator) [inline Backend/Services/MailService.cs:32]: Is it possible to connect only once? It looks like bunch of repetitive code in every method call

Comment (jacob-b-labs) [reply Backend/Services/MailService.cs:32]: AFAIK making an SMTP connection each time is not that resource heavy - I read that even Outlook sends messages by connecting a single time then closing the connection afterwards.

File: Backend/Controllers/HealthController.cs (hunk 1 above)
Comment (jacob-b-labs) [reply Backend/Controllers/HealthController.cs]: Fixed

Comment (jacob-b-labs) [reply Backend/Controllers/HealthController.cs]: Fixed

File: Backend/Controllers/HealthController.cs (hunk 2 above)
Comment (jacob-b-labs) [reply Backend/Controllers/HealthController.cs]: Fixed

File: Backend/Program.cs (hunk 3 above)
Comment (jacob-b-labs) [reply Backend/Program.cs]: Done. Had to check how to pass nested settings as environment variables.

**Pull request merged**